keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+
+  3. limits
+      * `pod.spec.resources.limits[memory/cpu] >= sum(pod.spec.containers[*].resources.limits[memory/cpu]) + sum(pod.spec.initContainers[*].resources.limits[memory/cpu])` is valid.
+
+          Pod-level resource limits are allowed to be greater than the sum of container level resource limits to account for ephemeral containers that could be added 
+          later. Setting a resource limit on the pod in its cgroup settings will not reserve any resource and hence not affect the other workloads running on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < sum(pod.spec.containers[*].resources.limits[memory/cpu]) + sum(pod.spec.initContainers[*].resources.limits[memory/cpu])` is valid.
+        
+        For example: consider a pod with multiple containers:
+          1. Container 1: Moderate CPU and memory usage, but occasional spikes.
+          2. Container 2: Low and consistent resource usage.
+
+
+          ```yaml
+          apiVersion: v1
+          kind: Pod
+          ...
+          spec:
+            resources:
+              requests:
+                memory: 100Mi
+                cpu: 1
+              limits:
+                memory: 400Mi
+                cpu: 2.5
+            containers:
+            - name: container-1
+                resources:
+                  requests:
+                    memory: 60Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 300Mi
+                    cpu: 2  
+            - name: container-2
+                resources:
+                  requests:
+                    memory: 40Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 200Mi
+                    cpu: 1 
+          ```
+
+          The pod-level limit will serve as an aggregate ceiling. This will cover the scenario where the sum of peak usage for individual containers in a pod is high, but these
+          peaks don’t necessarily occur simultaneously.
+
+          If we simply set the pod-level limit as the sum of the individual container limits, we might overallocate resources. The pod might never actually use all those resources
+          at once, leading to wasted capacity on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+  4. Validation logic comparing requests and limits
+
+      1. Pod level requests greater than pod level limits is invalid.
+
+      2. Pod level requests greater than sum of limits of all containers is invalid.
+
+      3. Sum of container level requests greater than Pod level limits is invalid.
+        
+          This is already ensured by other validation rules in 1. and 2. as following:
```

Comment:
those rules written in assumption that there is a pod level request defined. If pod level request is not defined, rules will not execute, so you need this another rule to check this condition.